client/src/components/student-data-form/StudentDataForm.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from "react";
+import { TextField, Button, Card, CardContent } from "@material-ui/core";
+import { StudentData } from "../../types/types";
+import { sendStudentData } from "../../api/api-utils";
+import "./StudentDataForm.css";
+import { Typography } from "@mui/material";
+
+const StudentDataForm: React.FC = () => {
```

Comment:
React.FC niepotrzebne
https://juhanajauhiainen.com/posts/should-you-use-reactfc-to-type-your-components
TLDR: No. You should probably not.